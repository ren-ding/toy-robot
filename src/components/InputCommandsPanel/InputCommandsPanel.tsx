import React, { Component } from 'react';
import './styles/InputCommandsPanel.css'

interface Props{
  onInputCommands:(event: {target: { value: string }}) => void
}

const InputCommands = ({
  onInputCommands
}:Props)=>(
  <textarea rows={20} className='input-commands-textarea' onChange={onInputCommands}/>
);

export default class InputCommandsPanel extends Component<Props> {
  constructor(props:Props) {
    super(props);
  }

  render() {
    return (
      <div className='input-commands-panel-wrapper'>
        <span className='input-label'>Please input commands</span>
        <InputCommands 
          onInputCommands={this.props.onInputCommands}
        />
      </div>
    );
  }
}
