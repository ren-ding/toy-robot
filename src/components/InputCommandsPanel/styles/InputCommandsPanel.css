.input-commands-panel-wrapper
{
    flex: 1;
    margin: 20px
}
  
.input-commands-panel-wrapper .input-label
{
    font-size: 25px;
    display: block;
    padding: 5px;
}

.input-commands-panel-wrapper .input-commands-textarea
{
    font-size: 20px;
    width: 300px;
}
