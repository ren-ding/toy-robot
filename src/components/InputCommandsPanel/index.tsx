import InputCommandsPanel from './InputCommandsPanel'
export default InputCommandsPanel