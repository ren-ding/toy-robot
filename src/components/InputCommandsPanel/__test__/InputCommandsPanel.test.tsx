import React from 'react';
import InputCommandsPanel from '../InputCommandsPanel'
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('InputPanel', ()=>{
  let inputCommandsPanel:any;
  let props:any;

  beforeEach(()=>{
    props = {
      onInputCommands: () => {}
    };
    inputCommandsPanel = () => mount(<InputCommandsPanel {...props} />)
  });

  describe('When initialized the component', ()=>{
    it('should render a span and an inputcommand textarea', () => {
      const panel = inputCommandsPanel();
      expect(panel.html()).toContain('Please input commands');
      expect(panel.find('InputCommands').length).toBe(1);
    })
  });
  
  describe('User typed in command textarea', ()=>{
    it('should call onInputCommands callback', () => {
      const mockCallback = jest.fn();
      props.onInputCommands = mockCallback;
      
      inputCommandsPanel().find('InputCommands').at(0).simulate('change');

      expect(mockCallback).toHaveBeenCalled();
      expect(mockCallback).toHaveBeenCalledTimes(1);
    })
  });
  
});