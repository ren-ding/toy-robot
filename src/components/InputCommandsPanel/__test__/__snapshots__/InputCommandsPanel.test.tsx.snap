// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPanel render should render to match with snapshot 1`] = `
<div
  className="input-commands-panel-wrapper"
>
  <span
    className="input-label"
  >
    Please input commands
  </span>
  <textarea
    className="input-commands-textarea"
    onChange={[Function]}
    rows={20}
  />
</div>
`;
